apps/marketing/src/app/(marketing)/blog/[post]/opengraph-image.tsx

```diff
@@ -0,0 +1,75 @@
+import { ImageResponse } from 'next/server';
+
+import { allBlogPosts } from 'contentlayer/generated';
+
+export const runtime = 'edge';
+
+export const size = {
+  width: 1200,
+  height: 630,
+};
+
+export const contentType = 'image/png';
+
+type BlogPostOpenGraphImageProps = {
+  params: { post: string };
+};
+
+export default async function BlogPostOpenGraphImage({ params }: BlogPostOpenGraphImageProps) {
+  const blogPost = allBlogPosts.find((post) => post._raw.flattenedPath === `blog/${params.post}`);
+
+  if (!blogPost) {
+    return null;
+  }
+
+  const [interBold, interRegular, backgroundImage, logoImage] = await Promise.all([
+    fetch(new URL('./../../../../assets/inter-bold.ttf', import.meta.url)).then(async (res) =>
+      res.arrayBuffer(),
+    ),
+    fetch(new URL('./../../../../assets/inter-regular.ttf', import.meta.url)).then(async (res) =>
+      res.arrayBuffer(),
+    ),
+    fetch(new URL('./../../../../assets/background-blog-og.png', import.meta.url)).then(
+      async (res) => res.arrayBuffer(),
+    ),
+    fetch(new URL('./../../../../../public/logo.png', import.meta.url)).then(async (res) =>
+      res.arrayBuffer(),
+    ),
```

Comment:
These still make me sad but there isn't much we can do I think

Reply:
How about if we kinda make it like,

```typescript
const assetUrls = [
  './../../../../assets/inter-bold.ttf',
  './../../../../assets/inter-regular.ttf',
  './../../../../assets/background-blog-og.png',
  './../../../../../public/logo.png',
];

async function fetchArrayBuffers(urls) {
  const responses = await Promise.all(urls.map(url => fetchArrayBuffer(url)));
  return responses;
}

const resourceArrayBuffers = await fetchArrayBuffers(assetUrls);
const [interBold, interRegular, backgroundImage, logoImage] = resourceArrayBuffers;
```

Reply:
That's still just fluff around what's happening, my sadness is mostly around needing a ton of `../../../` to get it working

Reply:
Yeah, I agree, though this make it less scary, the core issue still remains

Reply:
Did the tidy, broke generation of images due to the compiler intercepting these `import.meta.url` calls. Will leave it for now :smile: